frontend/src/components/SearchResult.js

```diff
@@ -5,22 +5,28 @@ import { IoIosAlarm } from 'react-icons/io';
 import { IoBarbellSharp } from 'react-icons/io5';
 import { useEffect, useState } from 'react';
 import { getRecipesWithMatchingIngredients } from './apiCalls';
+import ModalForSearch from './ModalForSearch';
 
 function SearchResult({ ingredients, searchBy }) {
-  const [recipe, setRecipe] = useState([]);
+  const [recipe, setRecipe] = useState(null);
+  const [show, setShow] = useState(false);
+  const handleShow = () => setShow(true);
 
   useEffect(() => {
     const getRecipes = async () => {
       const recipeList = await getRecipesWithMatchingIngredients(
         ingredients,
         searchBy
       );
-      recipeList
-        ? recipeList.sort((a, b) => a.name.localeCompare(b.name))
-        : console.log('Loading');
+      recipeList?.sort((a, b) => a.name.localeCompare(b.name));
+
       setRecipe(recipeList);
+      if (recipeList.length === 0) handleShow();
```

Comment:
Nem warningol a handleShow-ra, nem kell adatfüggőségként? Lehet nem, csak kérdezem.

Reply:
Inkább rakjuk {} közé, nem szerencsések az egysoros ifek, best practice kerülni őket:
http://www.fredosaurus.com/notes-java/flow/if/30if-braces.html

Komolyabb programhibák is eredtek már ebből:
https://www.imperialviolet.org/2014/02/22/applebug.html

Reply:
nincs warning ,  {} raktam